pkg/contracts/test/ProxyOwnerTest.t.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.19;
+
+import "forge-std/Test.sol";
+import {AlloSetup} from "allo-v2-test/foundry/shared/AlloSetup.sol";
+import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import {RegistryFactoryV0_0} from "../src/RegistryFactoryV0_0.sol";
+import {RegistryFactoryV0_1} from "../src/RegistryFactoryV0_1.sol";
+import {RegistryCommunityV0_0} from "../src/RegistryCommunityV0_0.sol";
+import {CVStrategyV0_0} from "../src/CVStrategyV0_0.sol";
+import {CollateralVault} from "../src/CollateralVault.sol";
+import {Upgrades} from "@openzeppelin/foundry/LegacyUpgrades.sol";
+import {ProxyOwner} from "../src/ProxyOwner.sol";
+import {ProxyOwnableUpgrader} from "../src/ProxyOwnableUpgrader.sol";
+
+contract ProxyOwnerTest is Test {
+    address deployerWallet = makeAddr("deployerWallet");
+    address anotherWallet = makeAddr("anotherWallet");
+    address protocolFeeReceiver = makeAddr("multisigReceiver");
+
+    function test_upgradeWithProxyAdmin() public {
+        vm.startPrank(deployerWallet);
+
+        ProxyOwner proxyOwner = ProxyOwner(
+            address(
+                new ERC1967Proxy(
+                    address(new ProxyOwner()), abi.encodeWithSelector(ProxyOwner.initialize.selector, deployerWallet)
```

Comment:
ah i see ProxyOwner its the contract that should implement the interface? so we need rename the interface to start with I

Reply:
its not an interface its a concrete contract that simply implement OwnableUpgreadable

Reply:
ProxyOwner is a concrete class